### PR TITLE
feat: add storybook preview to pr links

### LINKED ISSUE #4814: Add published Storybook link to top of PR
The link to view the deployed Storybook for viewing is hard to find on a new PR, so let's create a link at the top to make it easier for people to find/view it.

Screenshot 2024-01-16 at 1 39 51 PM

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8288850525), [Test report](https://leather-wallet.github.io/playwright-reports/feat/storybook-pr-link), [Storybook preview](https://65982789c7e2278518f189e7-olbgsatdye.chromatic.com/)

Add published Storybook preview link to top of pull requests.

### CODE REVIEW

File: .github/workflows/build-extension.yml [hunk 1]
```diff
@@ -73,7 +73,20 @@ jobs:
         run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
         id: extract_branch
 
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - run: yarn
```
Comment (kyranjamie) [inline .github/workflows/build-extension.yml]: Why is it required to checkout and run yarn?

If needed, would be better to use provision script, that has caching in place

File: .github/workflows/build-extension.yml [hunk 2]
```diff
@@ -73,7 +73,20 @@ jobs:
...
+
+      - name: Deploy to Chromatic
+        id: chromatic
+        uses: chromaui/action@latest
+        with:
+          autoAcceptChanges: 'main'
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (kyranjamie) [inline .github/workflows/build-extension.yml:81]: This deploy action is currently under the `set-download-link`.

Would be better to create a new step entirely (before set-download-link) called `deploy-storybook` with an output of `steps.chromatic.outputs.storybookUrl`. You'll then need to add that step to the `needs` array

File: .github/workflows/build-extension.yml [hunk 3]
```diff
@@ -73,7 +73,20 @@ jobs:
         run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
         id: extract_branch
 
+      - uses: actions/checkout@v3
```
Comment (kyranjamie) [inline .github/workflows/build-extension.yml]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by kyranjamie — CHANGES_REQUESTED** [review]
Great work @fbwoolf—this will be super helpful. 

Requesting changes only on grounds that it'd be a better Github action set up to have a separate step to do the deploy, rather than bundling in with the set download link job. Consider if the job fails, then we're more easily able to see which specific step failed. 

Can we also delete the `notify-release.yml` job as well this isn't needed anymore.

File: .github/workflows/build-extension.yml (hunk 2 above)
Comment (fbwoolf) [reply .github/workflows/build-extension.yml:81]: Ok, thx for feedback. I wasn't sure how to get the id/output to be recognized unless I put it in the step. I had a separate step first, but then the url wasn't correct. I see what you mean here with setting the output.

File: .github/workflows/build-extension.yml (hunk 1 above)
Comment (fbwoolf) [reply .github/workflows/build-extension.yml]: If I don't have this, then it fails. I'll try the provision script. 👍  It was part of the Chromatic docs example I followed.

**Comment by fbwoolf** [comment]
@kyranjamie hope this is ready now?

**Review by kyranjamie — APPROVED** [review]
Perfect, thanks @fbwoolf

**Pull request merged**